sources/packages/api/src/route-controllers/student-appeal/student-appeal.aest.controller.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Controller,
+  Param,
+  Get,
+  NotFoundException,
+  Patch,
+  Body,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+import { StudentAppealService, WorkflowActionsService } from "../../services";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, Groups, UserToken } from "../../auth/decorators";
+import { ApiTags, ApiUnprocessableEntityResponse } from "@nestjs/swagger";
+import BaseController from "../BaseController";
+import { ClientTypeBaseRoute } from "../../types";
+import { UserGroups } from "../../auth/user-groups.enum";
+import {
+  StudentAppealApiOutDTO,
+  StudentAppealApprovalApiInDTO,
+} from "./models/student-appeal.dto";
+import { getUserFullName } from "../../utilities";
+import { IUserToken } from "../../auth/userToken.interface";
+import {
+  STUDENT_APPEAL_INVALID_OPERATION,
+  STUDENT_APPEAL_NOT_FOUND,
+} from "../../services/student-appeal/constants";
+import { StudentAppealStatus } from "../../database/entities";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("appeal")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-appeal`)
+export class StudentAppealAESTController extends BaseController {
+  constructor(
+    private readonly studentAppealService: StudentAppealService,
+    private readonly workflowActionsService: WorkflowActionsService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the student appeal and its requests.
+   * @param appealId appeal id to be retrieved.
+   * @returns the student appeal and its requests.
+   */
+  @Get(":appealId/requests")
+  async getStudentAppealWithRequests(
+    @Param("appealId") appealId: number,
+  ): Promise<StudentAppealApiOutDTO> {
+    const studentAppeal =
+      await this.studentAppealService.getAppealAndRequestsById(appealId);
+    if (!studentAppeal) {
+      throw new NotFoundException("Not able to find the student appeal.");
```

Comment:
I think missed to add API tag of this exception. or will it be handled in next PRs

Reply:
That is a not NotFound when the id (part of the URL) is not found. IMO it falls under the global assumed ones.
My point is, in this situation, the message will not add much since it is the normal way that a NotFoun would happen.

Reply:
I get your point @andrewsignori-aot , but I feel when an exception is thrown we need to add it on the API tags, else we can just log the error and move on.

Reply:
I did not follow this part: "else we can just log the error and move on."

Reply:
No, I mean if we are not throwing an not found exception we may not have apitags

Reply:
ok, I goit it now. We definitely need to throw it 😉 
I just added the @ApiNotFound decorator.